docs-v2/design_proposals/buildx-transparent-support.md

```diff
@@ -0,0 +1,284 @@
+
+New Global Configs:
+
+* `buildx-builder`: Enables automatic detection of buildx and multiple builder support.
+* `cache-tag`: Allows overriding the default cache tagging strategy, useful for managing caches across different branches or environments.
+
+Skaffold Schema changes:
+
+* `cacheTo` to specify custom cache destinations, adding `--cache-to` support to the Docker CLI build process when using buildx.
+
+Build Process Enhancements:
+
+* The build process now intelligently detects and utilizes buildx if available and configured.
+* Reduced dependency on the local Docker daemon when using buildx, enhancing security and flexibility.
+
+Backward Compatibility:
+
+All changes are designed to be backward compatible.
+If buildx is not detected or used, Skaffold will fall back to the traditional Docker builder.
+
+This cache behavior is intended to provide transparent user experience, with similar functionality compared to traditional local docker builds,
+without additional boilerplate or different configuration / patches for a CI workflow.
+
+## Design approach
+
+`docker buildx build` can be configured to execute against a [remote buildkit instance](https://docs.docker.com/build/builders/drivers/remote/).
+No docker daemon is necessary for this to work, but also that is supported by default using the [docker driver](https://docs.docker.com/build/builders/drivers/docker/).
+Additionally, [docker container driver](https://docs.docker.com/build/builders/drivers/docker-container)
+or [kubernetes driver](https://docs.docker.com/build/builders/drivers/kubernetes/) are available too for advanced use cases.
+
+Since `docker build` and `docker buildx build` essentially share the same options, no major modifications are needed to Skaffold for backward compatibility.
+
+This proposal implements the logic to detect if buildx is the default builder (looking for an alias in the docker config).
+To [set buildx as the default builder](https://github.com/docker/buildx?tab=readme-ov-file#set-buildx-as-the-default-builder), the command `docker buildx install` should be used.
+
+Then, additional buildx features are availables, like multi-platform support and different cache destinations.
+
+Cache adjustment is extended via the the `cache-tag`, useful to point to latest or a generic cache tag, instead of the generated one for this build
+(via `tagPolicy` that would be useless in most distributed cases, as it can be invalidated by minor changes, specially if using `inputDigest`).
+
+Multi-platform images now can be built nativelly, without multiplexing the pipeline nor additional steps to stich the different images.
+
+### User experience
+
+To use BuildKit transparently, BuildX should be installed as default builder (this creates an alias in docker config):
+
+```
+docker buildx install
+```
+
+Then Skaffold should be configured to detect buildx (default builder) and set a generic cache tag:
```

Comment:
I believe, per below in the doc, these are optional. If my understanding is correct, it'd be great if this doc lists these as "can be" rather than "should be".